src/downshift.js

```diff
@@ -561,6 +563,15 @@ class Downshift extends Component {
 
   button_handleClick = event => {
     event.preventDefault()
+    // handle odd case for Safari and Firefox which
+    // don't give the button the focus properly.
+    /* istanbul ignore if (can't reasonably test this) */
+    if (
+      this.props.environment.document.activeElement ===
+      this.props.environment.document.body
```

Comment:
I'm using this if statement because some folks may want to automatically focus the input or some other thing when the button is pressed. This gets around the bug 👌

Reply:
Oh 😍😍😍